src/orca/services/sevenbridges/client_factory.py

```diff
@@ -0,0 +1,174 @@
+import os
+from dataclasses import dataclass, field
+from functools import cached_property
+from typing import Any, Optional
+
+from airflow.models.connection import Connection
+from sevenbridges import Api
+from sevenbridges.http.error_handlers import maintenance_sleeper, rate_limit_sleeper
+
+from orca.errors import ClientArgsError, ClientRequestError
+
+
+@dataclass
```

Comment:
The main advantage of the `dataclass` in the standard Python library is that it eliminates boilerplate (like defining an `__init__()` method that just stores the parameters as attributes. They intentionally avoid using the type hints for validation ([source](https://docs.python.org/3/library/dataclasses.html#class-variables)). 

That said, your comment prompted me to look into how easy it would be to leverage `pydantic` validation using type hints, and I found that they have a `dataclass` [drop-in replacement](https://docs.pydantic.dev/usage/dataclasses/). I've made the switch, so we're now validating the input values.